src/Framework/Constraint/Traversable/ArrayIsList.php

```diff
@@ -0,0 +1,64 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\Framework\Constraint;
+
+use function is_array;
+
+/**
+ * @no-named-arguments Parameter names are not covered by the backward compatibility promise for PHPUnit
+ */
+final class ArrayIsList extends Constraint
+{
+    /**
+     * Returns a string representation of the constraint.
+     */
+    public function toString(): string
+    {
+        return 'is list';
+    }
+
+    /**
+     * Evaluates the constraint for parameter $other. Returns true if the
+     * constraint is met, false otherwise.
+     */
+    protected function matches(mixed $other): bool
+    {
+        if (!is_array($other)) {
+            return false;
+        }
+
+        if ($other === []) {
+            return true;
+        }
+
```

Comment:
```suggestion

        if (function_exists('array_is_list')) {
            return array_is_list($other);
        }

```

On PHP 8.1, we have [a native function](https://www.php.net/array_is_list) for that. I suggest we leverage it if it's available.